gh-api: add page (#5127)
tldr-pages/tldr: pages/common/gh-api.md

# gh api

> Makes authenticated HTTP requests to the GitHub API and prints the response.
> More information: <https://cli.github.com/manual/gh_api>.

- Display the releases for the repository in the local path in JSON format:

`gh api repos/:owner/:repo/releases`

- Create a reaction for a specific issue:

> `gh api --header {{Accept:application/vnd.github.squirrel-girl-preview+json}} repos/:owner/:repo/issues/{{issue_number}}/reactions --raw-field {{content=+1}}`

Reviewer: How about this?
```suggestion
`gh api --header {{Accept:application/vnd.github.squirrel-girl-preview+json}} {{repos/:owner/:repo/issues/:issue_number/reactions}} --raw-field {{content=+1}}`
```

Reviewer: This also confuses me a bit&mdash;how do you specify which comment to react to?

Author: To create a reaction for an issue comment you should use [another endpoint](https://docs.github.com/en/free-pro-team@latest/rest/reference/reactions#create-reaction-for-an-issue-comment): `/repos/:owner/:repo/issues/comments/{{comment_id}}/reactions`

Author: Placeholder values like `:owner` and `:repo` in the endpoint get replaced with values from the repository of the current directory. That's why we can't use `:issue_number` because we're in conflict with the `gh api` syntax.
```suggestion
`gh api --header {{Accept:application/vnd.github.squirrel-girl-preview+json}} --raw-field '{{content=+1}}' {{repos/:owner/:repo/issues/123/reactions}}`
```

Reviewer: Ah okay, I see.

# gh api

> Makes authenticated HTTP requests to the GitHub API and prints the response.
> More information: <https://cli.github.com/manual/gh_api>.

- Display the subcommand help:

`gh api --help`

> - Display the releases for the repository in the local path in JSON format:

Author: ```suggestion
- Display the releases for the current repository in JSON format:
```

`gh api repos/:owner/:repo/releases`

- Create a reaction for a specific issue:

> `gh api --header {{Accept:application/vnd.github.squirrel-girl-preview+json}} --raw-field '{{content=+1}}' {{repos/:owner/:repo/issues/123/reactions}}`

Reviewer 2: Are you _sure_ that `Accept` header is right haha?

Reviewer: seems like it should be `Content-Type`...

Author: It's a developers preview since 2016 🤷‍♂️ 🐿️
https://docs.github.com/en/enterprise-server@2.20/rest/reference/reactions#delete-a-reaction
https://developer.github.com/changes/2016-05-12-reactions-api-preview/
